### PR TITLE
implement try_except decorator to ignore items/lines that raise

### PR DESCRIPTION
**Pull request by ganwell**
I just wanted you to know that I am trying to implement that.

There are no tests and no documentation, yet. I am not yet sure, that my approach is good. If you see that this is the wrong direction please let me know.

### CODE REVIEW

**Comment by ganwell** [comment]
@edk0 It seems you forgot to add lenses.py:

```
ImportError while importing test module '/home/ganwell/git/spy/tests/test_decorators.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_decorators.py:2: in <module>
    import lenses
E   ModuleNotFoundError: No module named 'lenses'
```

**Comment by edk0** [comment]
Thanks!

That's pretty much the way I was going to go, yeah, modulo style things. I'm not sure about returning None though. I feel like it should either return spy.DROP or a (somehow) settable default. Or maybe the exception itself?

**Comment by ganwell** [comment]
It seems that simple. The project is very nicely structured. I usually check the validity of the testcase, maybe you don't want that or you want it as an extra test. Do you need a cli/integration test?

File: doc/api/spy.decorators.rst [hunk 1]
```diff
@@ -92,6 +92,13 @@ This module contains various function decorators for use in spy fragments.
    Further work on this area is aspired to.
 
 
+.. decorator:: try_except
+
+   .. option:: --try, -t
+
+   Filter out input that causes the fragment to raise an exception. This is the
+   equivalent of a ``try: except:``-block in the fragment.
+
```
Comment (edk0) [inline doc/api/spy.decorators.rst:101]:
```suggestion


```

Comment (edk0) [inline doc/api/spy.decorators.rst:101]:
```suggestion


```

File: spy/decorators.py [hunk 2]
```diff
@@ -180,3 +180,11 @@ def apply(f, v):
 def magnify(fn, v, context, f):
     fn = partial(fn, context=context)
     return f(fn, v)
+
+@decorator("--try", "-t", doc="Filter out input that causes the fragment to raise an exception")
```
Comment (edk0) [inline spy/decorators.py:184]:
```suggestion


@decorator('--try', '-t', doc='Filter out input that causes the fragment to raise an exception')
```

File: tests/test_decorators.py [hunk 3]
```diff
@@ -198,3 +198,20 @@ def test(v):
         return v * 3
     l = list(spy.chain([test]).apply([[1,2,3], [4,5,6]]))
     assert l == [9, 18]
+
```
Comment (edk0) [inline tests/test_decorators.py:201]:
```suggestion


```

**Review by edk0 — CHANGES_REQUESTED** [review]
I have a couple of minor style nitpicks, but this looks pretty good. I'm happy with the test as written. As for integration testing, I don't think it's necessary since the decorator mechanism is pretty well covered. It might be nice to have an example in the docs (and then we could throw it in tests/regression_tests too because why not?), but I'm also happy to merge without.

**Review by edk0 — APPROVED** [review]

**Pull request merged**